shared-data/protocol/schemas/6.json

```
@@ -0,0 +1,1017 @@
+{
+  "$id": "opentronsProtocolSchemaV6",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+
+  "definitions": {
+    "pipetteName": {
+      "description": "Name of a pipette. Does not contain info about specific model/version. Should match keys in pipetteNameSpecs.json file",
+      "type": "string",
+      "enum": [
+        "p10_single",
+        "p10_multi",
+        "p20_single_gen2",
+        "p20_multi_gen2",
+        "p50_single",
+        "p50_multi",
+        "p300_single",
+        "p300_multi",
+        "p300_single_gen2",
+        "p300_multi_gen2",
+        "p1000_single",
+        "p1000_single_gen2"
+      ]
+    },
+
+    "moduleOnlyParams": {
+      "required": ["module"],
+      "additionalProperties": false,
+      "properties": {
+        "module": {
+          "type": "string"
+        }
+      }
+    },
+
+    "mmOffset": {
+      "description": "Millimeters for pipette location offsets",
+      "type": "number"
+    },
+
+    "offsetFromBottomMm": {
+      "description": "Offset from bottom of well in millimeters",
+      "required": ["offsetFromBottomMm"],
+      "properties": {
+        "offsetFromBottomMm": { "$ref": "#/definitions/mmOffset" }
+      }
+    },
+
+    "offset": {
+      "description": "x, y, and z component of offset vector in mm",
+      "required": ["x", "y", "z"],
+      "properties": {
+        "x": { "type": "number" },
+        "y": { "type": "number" },
+        "z": { "type": "number" }
+      }
+    },
+
+    "pipetteAccessParams": {
+      "required": ["pipette", "labware", "well"],
+      "properties": {
+        "pipette": {
+          "type": "string"
+        },
+        "labware": {
+          "type": "string"
+        },
+        "well": {
+          "type": "string"
+        }
+      }
+    },
+
+    "volumeParams": {
+      "required": ["volume"],
+      "volume": {
+        "type": "number"
+      }
+    },
+
+    "flowRate": {
+      "required": ["flowRate"],
+      "properties": {
+        "flowRate": {
+          "description": "Flow rate in uL/sec. Must be greater than 0",
+          "type": "number",
+          "minimum": 0
```

We can use `exclusiveMinimum` for this!

```suggestion
          "description": "Flow rate in uL/sec.",
          "type": "number",
          "exclusiveMinimum": 0
```